src/ceph-volume/ceph_volume/devices/lvm/batch.py

```diff
@@ -148,6 +150,31 @@ def execute(self, args):
 
         strategy.execute()
 
+    def _get_strategy(self, args):
+        strategy = get_strategy(args, args.devices)
+        unused_devices = [device for device in args.devices if not device.used_by_ceph]
+        # only data devices, journals can be reused
+        used_devices = [device.abspath for device in args.devices if device.used_by_ceph]
+        args.filtered_devices = {}
+        if used_devices:
+            for device in used_devices:
+                args.filtered_devices[device] = {"reasons": ["Used by ceph as a data device already"]}
+            if not args.report:
+                mlogger.info("Ignoring devices already used by ceph: %s" % ",".join(used_devices))
+        if not unused_devices and not args.report:
+            # report nothing changed
+            mlogger.info("All devices are already used by ceph. No OSDs will be created.")
+            raise SystemExit(0)
+        else:
+            new_strategy = get_strategy(args, unused_devices)
+            if type(strategy) != type(new_strategy):
+                if not args.report:
+                    mlogger.error("aborting because strategy changed from %s to %s after filtering" % (strategy, new_strategy))
+                    raise SystemExit(0)
```

Comment:
shouldn't this be a non-zero exit status?

Reply:
Yes, it probably should be.